internal/server/mcp.go

```diff
@@ -63,16 +65,27 @@ type mcpManager struct {
 	mcpSessions map[string]*mcpSession
 }
 
+func newMcpManager() *mcpManager {
+	mcpM := &mcpManager{
+		mu:          sync.RWMutex{},
+		mcpSessions: make(map[string]*mcpSession),
+	}
+	go mcpM.cleanupRoutine()
```

Comment:
There should be a way to close this so the gorountine doesn't leak (if the MCP manager shutdown for any reason).

Can you pass a context with cancel in here?